File: namastack-outbox-examples/namastack-outbox-example-java/src/main/java/io/namastack/demo/customer/Customer.java
package io.namastack.demo.customer;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.util.UUID;

@Entity(name = "customer")
public class Customer {

  @Id
  private UUID id;

  @Column(nullable = false)
  private String firstname;

  @Column(nullable = false)
  private String lastname;

  @Column(nullable = false)
  private String email;

  protected Customer() {
  }

  public Customer(UUID id, String firstname, String lastname, String email) {
    this.id = id;
    this.firstname = firstname;
    this.lastname = lastname;
    this.email = email;
  }

  public static Customer register(String firstname, String lastname, String email) {
    return new Customer(
        UUID.randomUUID(),
        firstname,
        lastname,
        email
    );
  }

  public UUID getId() {
    return id;
  }

  public String getFirstname() {
    return firstname;
  }

  public String getLastname() {
    return lastname;
  }

  public String getEmail() {
    return email;
  }
}

File: namastack-outbox-examples/namastack-outbox-example-jdbc/src/main/kotlin/io/namastack/demo/customer/Customer.kt
package io.namastack.demo.customer

import java.util.UUID

data class Customer(
    val id: UUID,
    val firstname: String,
    val lastname: String,
    val email: String,
) {
    companion object {
        fun register(
            firstname: String,
            lastname: String,
            email: String,
        ): Customer =
            Customer(
                id = UUID.randomUUID(),
                firstname = firstname,
                lastname = lastname,
                email = email,
            )
    }
}
